source/applications/advanced/color_balance.py

```diff
@@ -0,0 +1,203 @@
+"""
+Balance color for RGB image.
+"""
+
+import datetime
+import numpy as np
+import matplotlib.pyplot as plt
+import zivid
+
+
+def _display_rgb(rgb, title):
+    """Display RGB image.
+
+    Args:
+        rgb: RGB image
+        title: Image title
+
+    Returns None
+
+    """
+    plt.figure()
+    plt.imshow(rgb)
+    plt.title(title)
+    plt.show(block=False)
+
+
+def _set_settings(dimension, iris, exposure_time, brightness, gain):
+    """Set settings for capture (3D or 2D).
+
+    Args:
+        dimension: '3d' or '2d'
+        iris: Iris
+        exposure_time: Exposure time
+        brightness: Projector brightness
+        gain: Gain
+
+    Returns:
+        settings: Capture settings (3D or 2D)
+
+    """
+    if dimension == "3d":
+        settings = zivid.Settings()
+        settings.iris = iris
+        settings.exposure_time = datetime.timedelta(microseconds=exposure_time)
+        settings.brightness = brightness
+        settings.gain = gain
+    if dimension == "2d":
+        settings = zivid.Settings2D()
+        settings.iris = iris
+        settings.exposure_time = datetime.timedelta(microseconds=exposure_time)
+        settings.brightness = brightness
+        settings.gain = gain
+
+    return settings
+
+
+def _capture_rgb(camera, settings_2d):
+    """Capture 2D RGB image.
+
+    Args:
+        camera: Zivid camera
+        settings_2d: 2D capture settings
+
+    Returns:
+        rgb: RGB image
+
+    """
+    frame_2d = camera.capture_2d(settings_2d)
+    image = frame_2d.image().to_array()
+    rgb = np.dstack([image["r"], image["g"], image["b"], image["a"]])
+
+    return rgb
+
+
+def _compute_mean_rgb(rgb, pixels):
+    """Compute mean RGB values.
+
+    Args:
+        rgb: RGB image
+        pixels: Number of central pixels for computation
+
+    Returns:
+        mean_red: Red channel mean value
+        mean_green: Green channel mean value
+        mean_blue: Blue channel mean value
+
+    """
+    height = np.shape(rgb)[0]
+    width = np.shape(rgb)[1]
+    red = rgb[
+        int((height - pixels) / 2) : int((height + pixels) / 2),
+        int((width - pixels) / 2) : int((width + pixels) / 2),
+        0,
+    ]
+    green = rgb[
+        int((height - pixels) / 2) : int((height + pixels) / 2),
+        int((width - pixels) / 2) : int((width + pixels) / 2),
+        1,
+    ]
+    blue = rgb[
+        int((height - pixels) / 2) : int((height + pixels) / 2),
+        int((width - pixels) / 2) : int((width + pixels) / 2),
+        2,
+    ]
+    mean_red = np.mean(np.reshape(red, -1, []), dtype=np.float64)
+    mean_green = np.mean(np.reshape(green, -1, []), dtype=np.float64)
+    mean_blue = np.mean(np.reshape(blue, -1, []), dtype=np.float64)
+
+    return mean_red, mean_green, mean_blue
```

Comment:
What version of python is this sample targeted?

Reply:
I guess 3.5 or higher

Reply:
f-strings are python 3.6

Reply:
All our samples use f-strings. We never specify what version is required in the samples. Out tutorial 'setting up python' actually points at 3.7. Zivid-Python suggests 3.5 though. In any case, we should make an issue for the whole repository and not solve this for this PR.

Reply:
Sounds good. I think it is a dilemma, at one hand we want to use the latest and greatest from python, to show that we are trying our best to keep up to date and write the best code possible, but at the other hand we would also like samples to work with older python versions.

Reply:
The reason why I asked this was that we could return all of this information in one block:
```
@dataclass
class MeanColor:
    red: np.Array
    green: np.Array
    blue: np.Array
```
Which you could use like this:
`return MeanColor(red=mean_red, green=mean_green, blue=mean_blue)`

And later use it like this:
`mean_color.red`

Reply:
I've implemented this.